### ISSUE TITLE
Keep previous v3bw.txt files 

### ISSUE THREAD

**Issue by juga0** [pastly/simple-bw-scanner#179]
In https://trac.torproject.org/projects/tor/ticket/21378, the bwauths should keep the files used in their votes.

Currently sbws overwrites any previous ``v3bw.txt`` file when generating a new one.

I think it would be easier for the bwauths to keep the versions used in their votes if ``sbws`` while name the generated files with a timestamp, for instance ``20180605_1150_v3bw.txt`` (not sure which datetime name schema would be better here).

I'm not sure if the latest one should be a link, for instance ``v3bw.txt`` -> ``20180605_1150_v3bw.txt``.

Renaming oldest file according to the time they were generated would be more code complexity, so i don't think we should do it.

**Comment by pastly** [comment]
I hear you're going to work on this. Yay!

I think the config should have a path to a directory to store v3bw files in. For example, remove `v3bw_fname` and replace it with

```
[paths]
v3bw_dname = ${sbws_home}/v3bw
```

Then in there I'd write files with the current date and time in the name somehow (`YYYYMMDD_HHMM_v3bw.txt` seems fine to me). Finally, I'd have a `v3bw.txt` in there that is always a symlink to the newest file.

```
/home/user/.sbws/
    v3bw/
        v3bw.txt -> 20181010_0000_v3bw.txt
        20181010_0000_v3bw.txt
        20180909_0000_v3bw.txt
```

**Comment by juga0** [comment]
this is as i was doing right now:
   v3bw_dir = ${sbws_home}/v3bw 
   v3bw_fname = ${v3bw_dir}/v3bw_{}

The ``{}`` gets replaced by the timestamp.
i'm going to change ``v3bw_dir`` for `v3bw_dname`` and then would be mostly as you say except the date will be at the end.
Since we have a dir, maybe the name of the file could be just the timestamp as ``torflow``, and maybe i could format it also as ``torflow``: ``2018-06-13-05-10-00``?

**Comment by juga0** [comment]
Correct myself, ``torflow`` generates the files as ``bwscan.20180603-1645``, but it does not look nice, so i'd rather format the files as ``20180909_0000``.
would be this fine for you?

**Comment by pastly** [comment]
My new favorite option is `YYYYMMDD_HHMM.v3bw`

```
/home/user/.sbws/
    v3bw/
        latest.v3bw -> 20181010_0000.v3bw
        20181010_0000.v3bw
        20180909_0000.v3bw
```

But it is ultimately up to you and if I understand your last two comments correctly, yes your plan sounds fine.

**Comment by juga0** [comment]
ok, having a custom extension sounds good since is a custom format too.
Just thinking now i should also add the seconds, since one could run ``generate`` manually within seconds difference.

**Comment by pastly** [comment]
If you like my new favorite option of `YYYYMMDD_HHMMSS.v3bw`, then I don't think you need `v3bw_fname` in the config. You'd just need `v3bw_dname = ${sbws_home}/v3bw`. We won't be allowing as much flexibility to the user, but I think that's fine.

**Comment by juga0** [comment]
now i'm thinking it makes more sense to use the extension ``bwl110`` (bwl stand for Bandwidth List, as in the spec), even though the version is in the file.
Or even better ``bwl`` and rename all places where we use ``v3bw`` to it, then only add ``v110`` or ``v100`` when needed.

**Comment by juga0** [comment]
Though not sure is worth the change, and maybe i should leave the renaming for other issue

**Comment by juga0** [comment]
Well, i'm going to leave ``v3bw``

**Comment by pastly** [comment]
Yes I'd leave it.

**Comment by pastly** [comment]
Closed via #190
